libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java

```diff
@@ -691,13 +691,12 @@ private void bind() {
 
   private void initializeNavigationViewModel() {
     try {
-      if (getContext() instanceof ContextThemeWrapper) {
-        navigationViewModel = ViewModelProviders.of(
-                (FragmentActivity)(((ContextThemeWrapper) getContext()).getBaseContext()))
-                .get(NavigationViewModel.class);
-      } else {
-        navigationViewModel = ViewModelProviders.of((FragmentActivity) getContext()).get(NavigationViewModel.class);
+      Context context = getContext();
+      // unwrap the context if needed, see https://github.com/mapbox/mapbox-navigation-android/issues/2777
+      while (!(context instanceof FragmentActivity) && context instanceof ContextWrapper) {
```

Comment:
@LukasPaczos what's the while loop for? won't this get stuck instead of crash now?

Reply:
`Context` can be wrapped in multiple `ContextWrapper`s. We need to get to the bottom of this pile before trying to cast the base context. You can see that the loop will break if the `Context` is not a `ContextWrapper` anymore, so we won't be looping infinitely.

Reply:
Yeah makes sense, was concerned if someone makes a loop in their context wrapper. but, you'll have more issues if you do that. 

it's also in the [support library](https://android.googlesource.com/platform/frameworks/support/+/refs/heads/marshmallow-release/v7/mediarouter/src/android/support/v7/app/MediaRouteButton.java#265)